File: solutions/c/pangram/1/pangram.c
#include "pangram.h"
#include <ctype.h>
#include <stdbool.h>
#include <stddef.h>

bool is_pangram(const char *sentence) {
    if (sentence == NULL) {
        return false;
    }
    
    // Create a boolean array to track which letters we've seen
    bool letters_seen[26] = {false};
    
    // Iterate through each character in the sentence
    for (int i = 0; sentence[i] != '\0'; i++) {
        char c = sentence[i];
        
        // Convert to lowercase if it's a letter
        if (isalpha((unsigned char)c)) {
            c = tolower((unsigned char)c);
            letters_seen[c - 'a'] = true;
        }
    }
    
    // Check if all letters have been seen
    for (int i = 0; i < 26; i++) {
        if (!letters_seen[i]) {
            return false;
        }
    }
    
    return true;
}

File: solutions/javascript/pangram/1/pangram.js
export const isPangram = (sentence) => {
  // Return false for empty or too short sentences
  if (!sentence || sentence.length < 26) {
    return false;
  }
  
  // Create a Set to track unique letters
  const letters = new Set();
  
  // Convert to lowercase and iterate through each character
  const lowerSentence = sentence.toLowerCase();
  
  for (let i = 0; i < lowerSentence.length; i++) {
    const char = lowerSentence[i];
    
    // Check if it's a letter (a-z)
    if (char >= 'a' && char <= 'z') {
      letters.add(char);
      
      // Early exit: if we've found all 26 letters, return true
      if (letters.size === 26) {
        return true;
      }
    }
  }
  
  // Check if we have all 26 letters
  return letters.size === 26;
};